submissions/andrewklmn/friends-app/app.js

```diff
@@ -0,0 +1,302 @@
+  div.title = `Address: ${person.location.street.number}, ${person.location.street.name}, ${person.location.city}, ${person.location.state}
+  Phone: ${person.phone}
+  Email: ${person.email}`;
+  div.innerHTML =`
+      <div class="person-name ${person.gender}">${person.name.first} ${person.name.last}</div>
+      <div class="person-image">
+      <picture>
+        <source media="(max-width:300px)" 
+                srcset="${person.picture.thumbnail} 1x, ${person.picture.medium} 2x">
+        <source media="(max-width:400px)" 
+                srcset="${person.picture.medium} 1x, ${person.picture.large} 2x">
+        <img class="rounded" src="${person.picture.large}">
+      </picture>
+      </div>
+      <div class="person-age">
+        ${person.dob.age} y.o.
+      </div>
+      <div class="person-location">
+        ${person.location.country}
+      </div>
+  `;
+  container.appendChild(div);
+}
+
+const filterList = (friends)=>{
+  return friends.filter( person => {
+    if (state.filters.namePart != '') {
+      if((person.name.first + ' ' + person.name.last).toUpperCase()
+              .indexOf(state.filters.partOfName.toUpperCase()) == -1) { 
+        return false;
+      }
+    }
+    if (state.filters.gender != '') {
+      if(person.gender != state.filters.gender) { 
+        return false;
+      }
+    }
+
+    if (state.filters.ageRange[0] > person.dob.age
+        || state.filters.ageRange[1] < person.dob.age) {
+      return false;
+    }
+
+    if (state.filters.country != '') {
+      if(person.location.country != state.filters.country) { 
+        return false;
+      }
+    };
+
+    return true;
+  })
+}
+
+const sortList = (friends) => {
+
+  if(state.sorter.keyName == '') {
+    return friends;
+  }
+
+  const index0 = state.sorter.fieldPathList[state.sorter.keyName][0];
+  const index1 = state.sorter.fieldPathList[state.sorter.keyName][1];
+  
+  return friends.sort(function(a, b){
+    if(state.sorter.keyName == '') return 0;
+    if(state.sorter.order == 'DESC') {
+      if (a[index0][index1] > b[index0][index1]) {
+        return -1;
+      }
+      return 1;
+    }
+    if (a[index0][index1] < b[index0][index1]) {
+      return -1;
+    }
+    return 1;
+  });
+}
+
+const redrawFriends = (state) => {
+  preloader.classList.remove('hidden');
+  container.innerHTML = '';
+  state.numberOfShowedFriends = 0;
+  container.scrollTop = 0;
+
+  sortList(filterList(state.friends))
+    .slice(0, state.initialListLength)
+    .forEach(friend => {
+      state.numberOfShowedFriends++;
+      drawPerson(friend);
+    });
+
+  preloader.classList.add('hidden');
+}
+
+const updateFriendsList = (list)=> {
+  state.friends = list;
+  redrawFriends(state);
+}
+
+const addOptionToSelect = (select, optionValue, optionText) => {
+  const option = document.createElement('option');
+  option.value = optionValue;
+  option.innerHTML = optionText;
+  select.appendChild(option);
+}
+
+const drawSorter = (state)=>{
```

Comment:
I have destructured it in function body like this:
const {sorter} = state;
because I need the whole state object as an argument of redrawFriends(state) function call.